File: backend/app/schemas/documents.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class DocumentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    original_filename: str
    mime_type: str
    processing_status: str
    created_at: datetime
    updated_at: datetime


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total_count: int
    page: int
    page_size: int

File: frontend/src/types/documents.ts
export interface DocumentItem {
  id: string;
  original_filename: string;
  mime_type: string;
  processing_status: string;
  created_at: string;
  updated_at: string;
}

export interface DocumentListResponse {
  items: DocumentItem[];
  total_count: number;
  page: number;
  page_size: number;
}
